### ISSUE TITLE
Indentation of object destructuring default parameter

### ISSUE THREAD

**Issue by raxell** [eslint/eslint#10001]
**Tell us about your environment**

* **ESLint Version:** 4.18.1
* **Node Version:** 9.5.0
* **npm Version:** 5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint, but the same happens with default.

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "parser": "babel-eslint",
  "extends": "airbnb-base"
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
function x({
  a,
} = {
  a: 1, // Expected indentation of 4 spaces but found 2
}) { // Expected indentation of 2 spaces but found 0
  // ...
}
```

```bash
eslint file.js
```

**What did you expect to happen?**
No error should be detected. This only happens with function default parameters, if I do the same with variables the behaviour is as expected, this does not detect any error:
```js
const {
  a,
} = {
  a: 1,
};
```

**What actually happened? Please include the actual, raw output from ESLint.**
Eslint expects an additional level of indentation:
```
  4:1   error  Expected indentation of 4 spaces but found 2  indent
  5:1   error  Expected indentation of 2 spaces but found 0  indent
```

**Comment by platinumazure** [comment]
@not-an-aardvark Do you think this is a bug?

**Comment by not-an-aardvark** [comment]
The behavior described in the issue seems like it would be a bug, but I can't reproduce it in the [demo](https://eslint.org/demo/#eyJ0ZXh0IjoiLyogZXNsaW50IGluZGVudDogW2Vycm9yLCAyLCB7XG4gICAgICBTd2l0Y2hDYXNlOiAxLFxuICAgICAgVmFyaWFibGVEZWNsYXJhdG9yOiAxLFxuICAgICAgb3V0ZXJJSUZFQm9keTogMSxcbiAgICAgIEZ1bmN0aW9uRGVjbGFyYXRpb246IHtcbiAgICAgICAgcGFyYW1ldGVyczogMSxcbiAgICAgICAgYm9keTogMVxuICAgICAgfSxcbiAgICAgIEZ1bmN0aW9uRXhwcmVzc2lvbjoge1xuICAgICAgICBwYXJhbWV0ZXJzOiAxLFxuICAgICAgICBib2R5OiAxXG4gICAgICB9LFxuICAgICAgQ2FsbEV4cHJlc3Npb246IHtcbiAgICAgICAgYXJndW1lbnRzOiAxXG4gICAgICB9LFxuICAgICAgQXJyYXlFeHByZXNzaW9uOiAxLFxuICAgICAgT2JqZWN0RXhwcmVzc2lvbjogMSxcbiAgICAgIEltcG9ydERlY2xhcmF0aW9uOiAxLFxuICAgICAgZmxhdFRlcm5hcnlFeHByZXNzaW9uczogZmFsc2UsXG4gICAgICBpZ25vcmVkTm9kZXM6IFsnSlNYRWxlbWVudCcsICdKU1hFbGVtZW50ID4gKicsICdKU1hBdHRyaWJ1dGUnLCAnSlNYSWRlbnRpZmllcicsICdKU1hOYW1lc3BhY2VkTmFtZScsICdKU1hNZW1iZXJFeHByZXNzaW9uJywgJ0pTWFNwcmVhZEF0dHJpYnV0ZScsICdKU1hFeHByZXNzaW9uQ29udGFpbmVyJywgJ0pTWE9wZW5pbmdFbGVtZW50JywgJ0pTWENsb3NpbmdFbGVtZW50JywgJ0pTWFRleHQnLCAnSlNYRW1wdHlFeHByZXNzaW9uJywgJ0pTWFNwcmVhZENoaWxkJ10sXG4gICAgICBpZ25vcmVDb21tZW50czogZmFsc2VcbiAgICB9XSAqL1xuXG5mdW5jdGlvbiB4KHtcbiAgYSxcbn0gPSB7XG4gIGE6IDEsXG59KSB7XG4gIFxufSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6OSwic291cmNlVHlwZSI6Im1vZHVsZSIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==).

**Comment by raxell** [comment]
The problem seems solved with newer versions of eslint.
I don't receive the error anymore.

**Comment by not-an-aardvark** [comment]
Great, I'll go ahead and close this issue. Thanks!
